src/Data/Aeson/TextMap.hs

```diff
@@ -0,0 +1,210 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- |
+-- An abstract interface for maps from Textual keys to values.
+
+module Data.Aeson.TextMap (
+    TextMap,
```

Comment:
I see you wrote some haddocks, great!

It would be great to have sections like in https://hackage.haskell.org/package/containers-0.6.5.1/docs/Data-Map-Strict.html: Map type, construction, insertion, etc.